### DISCUSSION TITLE
Expose API Man on Public IP Address

### DISCUSSION THREAD

**Discussion by jtrugman** [apiman/apiman#2646] [Help]
I have been struggling to expose API Man to the public internet from my IP Addresses (instead of localhost) and running into some weird issues. 

I am using the [API Man Docker Compose Quick Start](https://www.apiman.io/apiman-docs/installation-guide/latest/quickstart.html) for version 3.1.3 Final

In the `.env` file I set 
```bash
APIMAN_HOSTNAME=MY_IP_ADDRESS
```
I then took down the container 
```bash
docker-compose down
```

and then restarted the container
```bash
docker-compose up -d
```
When I try to navigate to `http://my_ip_address:8080/apimanui`

I get redirected to `http://my_ip_address:8080/realms/apiman/protocol/openid-connect/auth?response_type=code&client_id=apimanui&redirect_uri=http%3A%2F%2Fmy_ip_address%3A8080%2Fapimanui&state=5b1b91be-3029-4859-9d8d-e674b68347c3&login=true&scope=openid` and a refused to connect error

When I go to the portal url I get a white screen and connection refused errors
![Screenshot 2024-06-07 at 3 26 31 PM](https://github.com/apiman/apiman/assets/28926820/f7e6f251-06ab-49a4-8702-bef25d69bec2)

It's not a Firewall issue:
- IP Address is publicly available
- Ports 8080 and 8081 are publicly available

I tried searching the docs and this forum but could not find anything to helpful. 

Has anyone else experienced this issue before / can someone please help me resolve it?

**Comment by msavy** [comment]
Hi, maintainer of Apiman here.

A small amount of investigative digging would be needed, as there can be several causes for this — usually security-related or Docker networking quirks (CORS, network, Keycloak tweaks, etc).

Are you a BPL support customer? If so, I'd be more than happy to spin up a support call for you to figure it out.

**Reply by msavy** [reply]
BTW, your redirect URL pasted into your post contains the IP address you were trying to redact, you might want to edit it (or let me know if you want me to redact on your behalf).

**Reply by jtrugman** [reply]
> BTW, your redirect URL pasted into your post contains the IP address you were trying to redact, you might want to edit it (or let me know if you want me to redact on your behalf).

Thank you! Just edited it

**Reply by jtrugman** [reply]
> Hi, maintainer of Apiman here.
> 
> A small amount of investigative digging would be needed, as there can be several causes for this — usually security-related or Docker networking quirks (CORS, network, Keycloak tweaks, etc).
> 
> Are you a BPL support customer? If so, I'd be more than happy to spin up a support call for you to figure it out.

Thanks @msavy! And not a BPL support customer yet, but definitely open to having a conversation about that.
